### ISSUE TITLE
[Accessibility] Audibly indicate that max locations selections have been reached

### ISSUE THREAD

**Issue by tallulahkay** [civiform/civiform#11921]
**Describe the bug**
When a user reaches the maximum number of allowed location selections, other location checkboxes are disabled, but there is no audible indication announced to screen reader users that this has occurred.

**WCAG Success Criterion**
[4.1.3 Status Messages (Level AA)](https://www.w3.org/TR/WCAG21/#status-messages)

**To Reproduce**
1. Enable screen reader (VoiceOver on Mac/iOS, NVDA/JAWS on Windows, TalkBack on Android)
2. Navigate to a location selection question with a maximum limit
3. Select locations until reaching the maximum
4. Attempt to navigate to or select another location
5. Observe that there is no announcement that maximum has been reached

**Expected behavior**
When the user reaches the maximum number of selections, the screen reader should announce this status change, such as "Maximum of X locations selected. Deselect a location to choose a different one."
